models/optional_modules/consent/snowplow_web_consent_totals.sql

```diff
@@ -0,0 +1,50 @@
+{{
+  config(
+    materialized='table',
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+with official_consent as (
+
+  select
+    consent_version as official_consent_version,
+    consent_scopes as official_consent_scopes,
+    consent_url as official_consent_url,
+    domains_applied as official_domains_applied,
+    derived_tstamp as official_start_date
+
+  from {{ ref('snowplow_web_consent_log') }}
+
+  where event_type = 'allow_all'
+
+  order by derived_tstamp desc
+
+  limit 1
+
+)
+
+, totals as (
+
+  select
+    count(*) as total_visitors,
+    count(case when last_consent_event_type ='allow_all' then 1 end) as allow_all,
+    count(case when last_consent_event_type ='allow_selected' then 1 end) as allow_selected,
+    count(case when last_consent_event_type IN ('allow_all', 'allow_selected') then 1 end) as allow,
+    count(case when last_consent_event_type = 'pending' then 1 end) as pending,
+    count(case when last_consent_event_type = 'deny_all'  then 1 end) as denied,
+    count(case when last_consent_event_type = 'expired'  then 1 end) as expired,
+    count(case when last_consent_event_type = 'withdrawn'  then 1 end) as withdrawn,
+    count(case when {{ dbt_utils.dateadd('year', '1', 'last_consent_event_tstamp') }} <= {{ dbt_utils.dateadd('month', '6', 'current_date') }}
+          and last_consent_event_type <> 'expired'
+          and {{ dbt_utils.dateadd('year', '1', 'last_consent_event_tstamp') }} > current_date then 1 end) as expires_in_six_months
+
+  from {{ ref('snowplow_web_consent_users') }}
+
+  where is_official_version
+
```

Comment:
❓ / 💡 Could there be some value in this table containing information about the previous versions as well? So the first query becomes a distinct and then the join is on the version? It might be useful to see how many users they still have on previous versions - or is that not something that makes sense in the context of cmps?

Reply:
As far as I understand whatever is the 'official', latest consent version that's all that matters from a compliance point of view. I can imagine from specific analytical purposes it could be useful to also see the previous version though so there's not much to lose by adding it there tbh.

Reply:
Makes sense, think it would be good to add if you can, after years of someone suddenly wanting to see some analysis without the filter _they_ asked for I'm just a little cautious around throwing away data that could be useful 😃